README.md
# Durrr(ability)
*Because everyone has had that Durrr moment and forgot to repair when we're able to.*
If you have an LDB display like [ChocolateBar](https://www.curseforge.com/wow/addons/chocolatebar) or [TitanPanel](https://www.curseforge.com/wow/addons/titan-panel) this will automatically place itself in your display.  Otherwise, it's just a minimap button that you can move around the minimap or hide completely if you want to.

## Durrr(ability) is here to see how borked your stuff is, and remind you to repair.
This began as a fork of [Broker_DurabilityInfo](https://www.curseforge.com/wow/addons/broker_durabilityinfo) in the Battle for Azeroth expansion as the addon had not yet been updated at that time, and over time I added features to it that I thought were missing and should be there.  Unfortunately, due to Shadowlands being less than stellar, I stepped away from the game and let my addons fall short.

I am now back for Dragonflight (and beyond?) and while credits go to [FulgerTunet](https://www.curseforge.com/members/fulgertunet) for his addon, I am updating and continuing to work on adding features to this as time goes on. If you don't need the additional features I've added, I highly suggest you look at using [Broker_DurabilityInfo](https://www.curseforge.com/wow/addons/broker_durabilityinfo) as it too has been updated for Dragonflight.

## Features
* Auto-Updating percentage display on your chosen LDB Display
* Show just the average percentage, or each item individually
* Choose to figure in the equipment in your bags into the average percentage or not
* Auto Repair
  * Choose to use Guild Funds (*If Available*)
    * Choose to use *ONLY* Guild Funds (*If Available*)
  * Choose whether or not to auto-repair based on your current reputation with a particular vendors
* Choose to have a Repair Warning pop up when you enter a "Rested" state (usually a city)
  * Set the threshold where this warning fires
* Choose to have a Repair Warning pop up when your durability falls below a threshold that you set
  * Set the threshold where this warning fires

### Misc
**Localizations have been enabled for this addon, but I do not speak any other language other than Broken-Redneck English. Any assistance in this regard will be very helpful, and I will give you credit for helping!**

[To Contribute](https://www.curseforge.com/wow/addons/durrrability/localization)

### About PapaBearDoes
I run a *501(c)(3)* dog rescue out of my house, so any and all donations to me are tax deductible. The rescue name is **K9 Trust**.

I am a founding member of [The Alphabet Cartel](https://discord.alphabetcartel.org), and we play on the ***\[US\]\[H\]Hellscream*** server under the guild ***&lt;Alphabet Cartel&gt;*** (we are looking for more players to join!).  We are a conglomeration of gamers and content creators that have banded together to provide a safe place for LGBTQIA+ members and allies, and to provide awareness and education on the issues that the Cartel experiences through games, discussion, and community.

You can join the Cartel via our [discord server](https://discord.alphabetcartel.org) (or in game, any member online can invite).

CHANGES.txt
lightweight tag a7f3efb405ed0a7f2093c28fae666ca86e8610d6 1.0.9
Author:	PapaBearDoes <73200436+PapaBearDoes@users.noreply.github.com>
Date:	Wed Dec 9 15:42:31 2020 -0700

commit a7f3efb405ed0a7f2093c28fae666ca86e8610d6
Author: PapaBearDoes <73200436+PapaBearDoes@users.noreply.github.com>
Date:   Wed Dec 9 15:42:31 2020 -0700

    Added Nag Options
    
    Added Nag Options and timers.
    Hopefully fixed issue #2.

commit 6953be424e21047cd8aef00fdf87c763e13a6cf4
Author: PapaBearDoes <73200436+PapaBearDoes@users.noreply.github.com>
Date:   Tue Dec 1 10:43:37 2020 -0700

    Update Durrrability.lua

commit 601756454ceddaaff01df1f1e429dbe398a1ab58
Author: PapaBearDoes <73200436+PapaBearDoes@users.noreply.github.com>
Date:   Mon Nov 23 03:50:50 2020 -0700

    Update Durrrability.toc

commit 715f849d00af0b37a82eee35a9f2d20db09e87aa
Author: PapaBearDoes <73200436+PapaBearDoes@users.noreply.github.com>
Date:   Fri Nov 20 16:35:30 2020 -0700

    Update Durrrability.lua

commit 67a8a3cf7ad6baf82f4e53f68a93bf754266bbb7
Author: PapaBearDoes <73200436+PapaBearDoes@users.noreply.github.com>
Date:   Fri Nov 20 13:39:17 2020 -0700

    Update Durrrability.lua



defaults.lua
--[[
                                      \\\\///
                                     /       \
                                   (| (.)(.) |)
     .---------------------------.OOOo--()--oOOO.---------------------------.
     |                                                                      |
     |  @file-author@'s Durrrability Addon for World of Warcraft
     ######################################################################## ]]
--   ## Let's init this file shall we?
-- Imports
local _G = _G
--Durrr = select(2, ...)
local me, ns = ...
local addon = LibStub("LibInit"):NewAddon(ns, me, true, "AceConsole-3.0", "AceEvent-3.0", "AceTimer-3.0")
local L = addon:GetLocale()
-- End Imports
--[[ ######################################################################## ]]
--   ## Do All The Things!!!
-- Define Globals
Durrr_globals = {
  enableTasks = {},
  ID = 6,
  SLOT = 4,
  NAME = 5,
  VAL = 1,
  MAX = 2,
  COST = 3,
  repairAllCost = nil,
  canRepair = false,
  combatState = false,
  vendorState = false,
  updateReq = true,
  slots = {
    {0, 0, 0, "Head", L["Head"], 0},
    {0, 0, 0, "Neck", L["Neck"], 0},
    {0, 0, 0, "Shoulder", L["Shoulder"], 0},
    {0, 0, 0, "Back", L["Back"], 0},
    {0, 0, 0, "Chest", L["Chest"], 0},
    {0, 0, 0, "Wrist", L["Wrist"], 0},
    {0, 0, 0, "Hands", L["Hands"], 0},
    {0, 0, 0, "Waist", L["Waist"], 0},
    {0, 0, 0, "Legs", L["Legs"], 0},
    {0, 0, 0, "Feet", L["Feet"], 0},
    {0, 0, 0, "MainHand", L["MainHand"], 0},
    {0, 0, 0, "SecondaryHand", L["SecondaryHand"], 0},
  },
  repColor = {
    [4] = "yellow",
    [5] = "lime",
    [6] = "00ff88",
    [7] = "00ffcc",
    [8] = "cyan",
  },
}

-- Create any required hidden frames
Durrr_frame = CreateFrame("GameTooltip")
Durrr_frame:SetOwner(WorldFrame, "ANCHOR_NONE")

-- Create DB defaults
Durrr_dbDefaults = {
  profile = {
    showDetails = true,
    showBags = true,
    updateInCombat = false,
    repairFromGuild = true,
    repairFromGuildOnly = false,
    repairThreshold = 4,
    showPopup = true,
    repairType = 1,
    alwaysAsk = false,
    warntoRepair = true,
    warnThreshold = 65,
    critWarntoRepair = true,
    critWarnThreshold = 25,
    modules = {
      ["*"] = 3
    }
  }
}

--Create config defaults
Durrr_options = {
  type = "group",
  name = "Durrrability",
  args = {
    general = {
      order = 1,
      type = "group",
      name = L["GeneralSettings"],
      cmdInline = true,
      args = {
        separator1 = {
          type = "header",
          name = L["DisplayOptions"],
          order = 1,
        },
        details = {
          order = 2,
          type = "toggle",
          name = L["ShowEachItem"],
          desc = L["ShowAllItemsToggle"],
          get = function()
            return addon.db.profile.showDetails
          end,
          set = function(key, value)
            addon.db.profile.showDetails = value
          end,
        },
        bags = {
          order = 3,
          type = "toggle",
          name = L["ShowBags"],
          desc = L["ShowBagsToggle"],
          get = function()
            return addon.db.profile.showBags
          end,
          set = function(key, value)
            addon.db.profile.showBags = value
          end,
        },
        combat = {
          order = 4,
          type = "toggle",
          name = L["InCombatUpdate"],
          desc = L["InCombatToggle"],
          get = function()
            return addon.db.profile.updateInCombat
          end,
          set = function(key, value)
            addon.db.profile.updateInCombat = value
          end,
        },
        separator3 = {
          type = "header",
          name = L["RepOpts"],
          order = 5,
        },
        factionThreshold = {
					order = 6,
					type = "select",
					style = "dropdown",
					name = L["MinRep"],
					desc = L["MinRepLevel"],
					get = function()
						return addon.db.profile.repairThreshold
					end,
					set = function(key, value)
						addon.db.profile.repairThreshold = value
					end,
					values = function()
						return {
							[4] = _G["FACTION_STANDING_LABEL4"],
							[5] = _G["FACTION_STANDING_LABEL5"],
							[6] = _G["FACTION_STANDING_LABEL6"],
							[7] = _G["FACTION_STANDING_LABEL7"],
							[8] = _G["FACTION_STANDING_LABEL8"],
						}
					end,
					disabled = function()
						return addon.db.profile.repairType == 0
					end,
				},
				askEverywhere = {
					order = 7,
					type = "toggle",
          width = "full",
					name = L["AskIfLower"],
					desc = L["LowRepConfirmPop"],
					get = function()
						return addon.db.profile.alwaysAsk
					end,
					set = function(key, value)
						addon.db.profile.alwaysAsk = value
					end,
					disabled = function()
						return addon.db.profile.repairType == 0
					end,
				},
        separator2 = {
          type = "header",
          name = L["RepairOpts"],
          order = 8,
        },
        repairType = {
          order = 9,
          type = "select",
          style = "dropdown",
          name = L["RepairType"],
          desc = L["VendorRepairQuestion"],
          get = function()
            return addon.db.profile.repairType
          end,
          set = function(key, value)
            addon.db.profile.repairType = value
            addon:UpdateIcon()
          end,
          values = function()
            return {
              [0] = L["DoNothing"],
              [1] = L["AutoRepair"],
              [2] = L["Ask"],
            }
          end,
        },
        repairGuild = {
          order = 10,
          type = "toggle",
          width = "full",
          name = L["UseGuildFunds"],
          desc = L["GuildFundsToggle"],
          get = function()
            return addon.db.profile.repairFromGuild
          end,
          set = function(key, value)
            addon.db.profile.repairFromGuild = value
            addon:UpdateIcon()
          end,
          disabled = function()
            return not (addon.db.profile.repairType == 1)
          end,
        },
        repairGuildOnly = {
          order = 11,
          type = "toggle",
          width = "full",
          name = L["OnlyGuildFunds"],
          desc = L["NoGuildGoldToggle"],
          get = function()
            return addon.db.profile.repairFromGuildOnly
          end,
          set = function(key, value)
            addon.db.profile.repairFromGuildOnly = value
          end,
          disabled = function()
            return (addon.db.profile.repairFromGuild == false or (not (addon.db.profile.repairType == 1)))
          end,
        },
				separator4 = {
					type = "header",
					name = L["WarningOpts"],
					order = 12,
				},
				warn = {
					order = 13,
					type = "toggle",
					name = L["CityWarnConf"],
					desc = L["CityWarnToggle"],
					get = function()
						return addon.db.profile.warntoRepair
					end,
					set = function(key, value)
						addon.db.profile.warntoRepair = value
						if (value) then
							addon:RegisterEvent("PLAYER_UPDATE_RESTING", "OnWarnUpdate")
						else
							addon:UnregisterEvent("PLAYER_UPDATE_RESTING")
						end
					end,
				},
				warnThreshold = {
					order = 14,
					type = "range",
					name = L["WarnThreshold"],
					desc = L["WarnMax"],
					min = 0, max = 100, step = 1,
					get = function()
						return addon.db.profile.warnThreshold
					end,
					set = function(key, value)
						addon.db.profile.warnThreshold = value
					end,
        },
        separator5 = {
					type = "header",
					name = L["CritWarningOpts"],
					order = 15,
				},
				critWarn = {
					order = 16,
					type = "toggle",
          width = "full",
					name = L["CritWarnConf"],
					desc = L["CritWarnToggle"],
					get = function()
						return addon.db.profile.critWarntoRepair
					end,
					set = function(key, value)
						addon.db.profile.critWarntoRepair = value
						if (value) then
              addon:RegisterEvent("ZONE_CHANGED", "OnWarnUpdate")
              addon:RegisterEvent("ZONE_CHANGED_NEW_AREA", "OnWarnUpdate")
						else
              addon:RegisterEvent("ZONE_CHANGED")
              addon:RegisterEvent("ZONE_CHANGED_NEW_AREA")
						end
					end,
				},
				critWarnThreshold = {
					order = 17,
					type = "range",
					name = L["WarnThreshold"],
					desc = L["WarnMax"],
					min = 0, max = 100, step = 1,
					get = function()
						return addon.db.profile.critWarnThreshold
					end,
					set = function(key, value)
						addon.db.profile.critWarnThreshold = value
					end,
        },
      },
    },
  },
}
--[[
     ########################################################################
     |  Last Editted By: @file-author@ - @file-date-iso@
     |  @file-revision@
     |                                                                      |
     '-------------------------.oooO----------------------------------------|
                              (    )     Oooo.
                              \  (     (    )
                               \__)     )  /
                                       (__/                                   ]]
